torchqdynamics/solver_utils.py

```diff
@@ -1,20 +1,21 @@
 import torch
 
 
-def kraus_map(rho: torch.Tensor, operators: torch.Tensor) -> torch.Tensor:
+def kraus_map(rho: torch.Tensor, O: torch.Tensor) -> torch.Tensor:
     """Compute the application of a Kraus map on an input density matrix.
 
     This is equivalent to `torch.sum(operators @ rho[None,...] @ operators.adjoint(),
     dim=0)`. The use of einsum yields better performances on large matrices, but may
     cause a small overhead on smaller matrices (N <~ 50).
 
     Args:
-        rho: Density matrix of shape (..., n, n).
-        operators: Kraus operators of shape (b, n, n).
+        rho: Density matrix of shape (a, ..., n, n).
+        operators: Kraus operators of shape (a, b, n, n).
     Returns:
-        Density matrix of shape (..., n, n) with the Kraus map applied.
+        Density matrix of shape (a, ..., n, n) with the Kraus map applied.
     """
-    return torch.einsum('mij,...jk,mkl->...il', operators, rho, operators.adjoint())
+    # TODO: fix doc
```

Comment:
Once you agree overall I will fix the doc (specify what are `a` and `b`, add the fact that the kraus map is not normalized).

Reply:
Ho and the renaming of `operators` to `O` is a proposed new convention: when objects are batched we don't use the plural or prefix them with `b` or `batched`, we just use the singular regular name (like we do for `rho`) and the doc tells what's the size. I also propose to use `O` for any generic operator argument.

Reply:
Ok i'm fine with this convention.